extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/taskadapter/PodTemplateTaskAdapter.java

```diff
@@ -126,7 +131,17 @@ public PodTemplateTaskAdapter(
   @Override
   public Job fromTask(Task task) throws IOException
   {
-    PodTemplate podTemplate = templates.getOrDefault(task.getType(), templates.get("base"));
+    PodTemplate podTemplate = null;
+    ExecutionConfig executionConfig = executionConfigRef.get();
+    if (executionConfig != null && executionConfig.getBehaviorStrategy() != null) {
+      String category = executionConfig.getBehaviorStrategy().getTaskCategory(task);
+      if (category != null) {
+        podTemplate = templates.get(category);
+      }
+    }
+    if (podTemplate == null) {
+      podTemplate = templates.getOrDefault(task.getType(), templates.get("base"));
```

Comment:
It would be cool to make this another kind of behavior, and have it be default. That would be clean